src/Adapter/Module/CommandHandler/BulkToggleModuleStatusHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Module\CommandHandler;
+
+use Module;
+use PrestaShop\PrestaShop\Core\Domain\Module\Command\BulkToggleModuleStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Module\CommandHandler\BulkToggleModuleStatusHandlerInterface;
+
+/**
+ * Bulk toggles Module status
+ */
+class BulkToggleModuleStatusHandler implements BulkToggleModuleStatusHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(BulkToggleModuleStatusCommand $command): void
+    {
+        foreach ($command->getModules() as $moduleName) {
+            $module = Module::getInstanceByName($moduleName);
+            if (!$module) {
+                continue;
+            }
+            if ($command->getExpectedStatus()) {
+                $module->enable();
+            } else {
+                $module->disable();
+            }
+        }
```

Comment:
Be careful, it seems to me that in prod environment the kernel caches the list of activated modules So after this loop you need to clear the caches (especially Symfony one but.it's safer to clear them all), if you don't the modification of the status will be ignored even if the state of the DB has changed.

In the ModuleManager class the cache clear is automatically performed when necessary:
https://github.com/PrestaShop/PrestaShop/blob/7c1f1dca6e8e079534aefa118846b2c1ec78de35/src/Core/Addon/Module/ModuleManager.php#L687-L692

But since here you use `Module::disabled/enable` "manualy" you need to handle the cache clear your self Either that or you need to use the `ManagerBuilder` in this handler (which might not be such a bad idea since it handles additional internal checking, and it's used by the module manager and the module CLI command already)